### PR TITLE
[redtube] fix #1663

### LINKED ISSUE #1663: RedTube DL don't work any more
### Checklist

- [X] I'm reporting a broken site
- [X] I've verified that I'm running yt-dlp version **2021.11.10.1**. ([update instructions](https://github.com/yt-dlp/yt-dlp#update))
- [X] I've checked that all provided URLs are alive and playable in a browser
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/ytdl-org/youtube-dl#video-url-contains-an-ampersand-and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues including closed ones. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

_No response_

### Description

only downloaded a small file (893 btyes) which is obviously not the real video file...
sample url:
https://www.redtube.com/39016781

### Verbose log

```shell
Download Log for url: https://www.redtube.com/39016781

Waiting
[RedTube] 39016781: Downloading webpage
[RedTube] 39016781: Downloading m3u8 information
[info] 39016781: Downloading 1 format(s): 1
[download] Destination: C:\Users\Sony\xxx\Youtube-Dl\Cute Asian is fucked & creampied!-39016781.mp4
[download] 893.00B at 436.07KiB/s (00:02)
[download] 100% of 893.00B in 00:02
Download Worker Finished.
```

### PR DESCRIPTION
**Pull request by nixxo**
### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [x] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [x] Improvement

---

### Description of your *pull request* and other information
Lonely day ;-)

Redtube seems to send json instead of m3u8 file so it requires an extra redirection.

Fix found in youtube-dl by user @dirkf and adapted
https://github.com/ytdl-org/youtube-dl/pull/29318

### CODE REVIEW

File: yt_dlp/extractor/redtube.py [hunk 1]
```diff
@@ -84,24 +87,33 @@ def _real_extract(self, url):
...
+            format_url = url_or_none(media.get('videoUrl'))
+            if not format_url:
+                continue
+            format_id = media.get('format')
+            quality = media.get('quality')
+            if format_id == 'hls' or (format_id == 'mp4' and not quality):
+                more_media = self._download_json(format_url, video_id, fatal=False)
+            else:
+                more_media = [media]
+            for media in more_media if isinstance(more_media, list) else []:
```
Comment (pukkandan) [inline yt_dlp/extractor/redtube.py:100]: any reason for the isinstance check? Cant this just be `more_media or []`? Same above

File: yt_dlp/extractor/redtube.py [hunk 2]
```diff
@@ -84,24 +87,33 @@ def _real_extract(self, url):
...
-                    'format_id': format_id,
-                    'height': int_or_none(format_id),
-                })
+                else:
+                    format_id = media.get('quality')
+                    formats.append({
+                        'url': format_url,
+                        'format_id': format_id,
+                        'height': int_or_none(format_id),
+                    })
```
Comment (pukkandan) [inline yt_dlp/extractor/redtube.py:116]: unnecessary change

File: yt_dlp/extractor/redtube.py (hunk 1 above)
Comment (nixxo) [reply yt_dlp/extractor/redtube.py:100]: I guess is a copy of line 90...

File: yt_dlp/extractor/redtube.py (hunk 2 above)
Comment (nixxo) [reply yt_dlp/extractor/redtube.py:116]: yup

File: yt_dlp/extractor/redtube.py (hunk 1 above)
Comment (dirkf) [reply yt_dlp/extractor/redtube.py:100]: If `more_media` was parsed from JSON (a value returned by `json.loads()`), we don't know that it's iterable: it might be `True`, or `42`.

File: yt_dlp/extractor/redtube.py (hunk 2 above)
Comment (dirkf) [reply yt_dlp/extractor/redtube.py]: I think that's what dstftw said too. The argument is that the `else` clause prevents accidentally breaking the logic if someone changes the `continue` at the end of the `if` suite. The codebase does tend to the original style.
